src/View/Helper/PhoneNumberFormat.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\I18n\PhoneNumber\View\Helper;
+
+use Laminas\I18n\PhoneNumber\CountryCode;
+use Laminas\I18n\PhoneNumber\Exception\ExceptionInterface;
+use Laminas\I18n\PhoneNumber\PhoneNumberValue;
+
+final class PhoneNumberFormat
+{
+    private CountryCode $country;
+
+    public function __construct(CountryCode $defaultCountryCode)
+    {
+        $this->country = $defaultCountryCode;
+    }
+
+    public function __invoke(): self
+    {
+        return $this;
+    }
+
+    /**
+     * @param non-empty-string|null $countryCodeOrLocale
+     */
+    private function coalesceCountryCode(?string $countryCodeOrLocale): CountryCode
+    {
+        $countryCodeOrLocale = $countryCodeOrLocale ? CountryCode::tryFromString($countryCodeOrLocale) : null;
+
+        return $countryCodeOrLocale ?: $this->country;
```

Comment:
If the values are always `CountryCode|null`, let's use `??` instead of `?:` (which does implicitly cast)